lib/Agrammon/Web/Routes.pm6

```diff
@@ -305,36 +305,38 @@ sub api-routes (Str $schema, $ws) {
         }
         operation 'exportExcel', -> LoggedIn $user {
             request-body -> %params {
+                # prevent header injection
+                my $filename = "%params<datasetName>.subst(/<-[\w\s_.-]>/, '', :g).xlsx";
+                my $excel = $ws.get-excel-export($user, %params).to-blob;
                 response.append-header(
```

Comment:
There's a helper `header` so you can just write this as `header 'Content-disposition', ...;` instead of the `response.append-header` call (they are equivalent, just the first is a little shorter).

Reply:
Done.